src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.cc

```diff
@@ -227,18 +227,25 @@ void WorkStealingThreadPool::PostforkChild() { pool_->Postfork(); }
 
 WorkStealingThreadPool::WorkStealingThreadPoolImpl::WorkStealingThreadPoolImpl(
     size_t reserve_threads)
-    : reserve_threads_(reserve_threads), queue_(this), lifeguard_(this) {}
+    : reserve_threads_(reserve_threads), queue_(this) {}
+
+WorkStealingThreadPool::WorkStealingThreadPoolImpl::
+    ~WorkStealingThreadPoolImpl() {
+  grpc_core::MutexLock lock(&lifeguard_ptr_mu_);
```

Comment:
This should be unnecessary, since the program will crash if Quiesce is not called before destruction, and Quiesce resets the lifeguard_.

Reply:
Would it make sense to add assert? Just in case.

Reply:
See https://github.com/grpc/grpc/blob/641dbfd4c24be9262b1b49a798e7f8116819d820/src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.cc#L186-L188

Reply:
Done